confmap/confmaptest/configtest.go

```diff
@@ -15,17 +15,46 @@
 package confmaptest // import "go.opentelemetry.io/collector/confmap/confmaptest"
 
 import (
-	"context"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"regexp"
+
+	"gopkg.in/yaml.v2"
 
 	"go.opentelemetry.io/collector/confmap"
-	"go.opentelemetry.io/collector/confmap/provider/fileprovider"
 )
 
 // LoadConf loads a confmap.Conf from file, and does NOT validate the configuration.
 func LoadConf(fileName string) (*confmap.Conf, error) {
-	ret, err := fileprovider.New().Retrieve(context.Background(), "file:"+fileName, nil)
+	// Clean the path before using it.
+	content, err := ioutil.ReadFile(filepath.Clean(fileName))
 	if err != nil {
+		return nil, fmt.Errorf("unable to read the file %v: %w", fileName, err)
+	}
+
+	var rawConf map[string]interface{}
+	if err = yaml.Unmarshal(content, &rawConf); err != nil {
 		return nil, err
 	}
-	return ret.AsConf()
+
+	return confmap.NewFromStringMap(rawConf), nil
+}
```

Comment:
I had to avoid using `fileprovider.New()` to avoid circular dependencies, since `fileprovider` uses `ValidateProviderScheme`.